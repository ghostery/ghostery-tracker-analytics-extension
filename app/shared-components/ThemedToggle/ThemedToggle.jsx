import React from 'react';
import PropTypes from 'prop-types';
import ClassNames from 'classnames';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Tooltip from 'react-bootstrap/Tooltip';

import './ThemedToggle.scss';

const ThemedToggle = ({
  children,
  activeButton,
  handleClick,
  icons,
  tooltipPosition,
  tooltipStrings,
  tooltipRef,
}) => (
  <div className="ThemedToggle d-flex">
    {children.map((child, i) => {
      const button = ClassNames(
        'ThemedToggle__button',
        {
          clicked: i === activeButton,
          icons,
          'd-flex': icons,
          'align-items-center': icons,
          'justify-content-center': icons,
        },
      );
      if (!tooltipStrings || !tooltipRef) {
        return (
          <div
            className={button}
            onClick={() => handleClick(i)}
            key={`tooltip-sans-${child.key ? child.key : child.props.children}`}
          >
            {children[i]}
          </div>
        );
      }
      return (
        <OverlayTrigger
          key={`overlay-${tooltipStrings[i]}`}
          placement={tooltipPosition}
          container={tooltipRef}
          overlay={(<Tooltip className="tooltipLeft">{tooltipStrings[i]}</Tooltip>)}
          delay={{ show: 500 }}
        >
          <div className={button} onClick={() => handleClick(i)}>
            {children[i]}
          </div>
        </OverlayTrigger>
      );
    })}
  </div>
);

ThemedToggle.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
  activeButton: PropTypes.number.isRequired,
  handleClick: PropTypes.func.isRequired,
  icons: PropTypes.bool,
  tooltipPosition: PropTypes.oneOf(['top', 'bottom', 'left', 'right']),
  tooltipStrings: PropTypes.arrayOf(PropTypes.string),
  tooltipRef: PropTypes.instanceOf(PropTypes.element),
};

ThemedToggle.defaultProps = {
  icons: false,
  tooltipPosition: null,
  tooltipStrings: null,
  tooltipRef: null,
};

export default ThemedToggle;
