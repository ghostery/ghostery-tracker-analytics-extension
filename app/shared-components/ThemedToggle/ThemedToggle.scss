@import '../../utils/scss/color-variables.scss';

.ThemedToggle {
  p {
    margin: 0;
    font-size: 10px;
  }
}

.ThemedToggle__button {
  background-color: $silver;
  border: 1px solid $silver;
  padding: 5px 8px 6px;
  width: 30px;
  height: 13px;
  color: $dark-grey;
  cursor: pointer;

  &:first-of-type {
    border-top-left-radius: 3px;
    border-bottom-left-radius: 3px;
    padding-left: 9px;
    padding-right: 7px;
  }

  &:last-of-type {
    border-top-right-radius: 3px;
    border-bottom-right-radius: 3px;
    padding-left: 7px;
    padding-right: 9px;
  }

  &.clicked {
    box-shadow: inset 1px 0 6px $opaque-background-medium;
    background-color: $white;
    border: 1px solid $white;
    color: $ghosty-blue;
  }

  &.icons {
    padding: 6px 8px 5px;
    height: 8px;
  }

  p {
    text-align: center;
  }
}
