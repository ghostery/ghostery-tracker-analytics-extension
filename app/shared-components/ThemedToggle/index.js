import { connect } from 'react-redux';
import ThemedToggle from './ThemedToggle.jsx';

const mapStateToProps = state => ({
  tooltipRef: state.UserInterfaceReducer.tooltipRef,
});

export default connect(mapStateToProps)(ThemedToggle);
