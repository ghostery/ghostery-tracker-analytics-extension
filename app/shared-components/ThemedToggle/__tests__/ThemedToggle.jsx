import React from 'react';
import { shallow } from 'enzyme';
import path from 'path';
import fakeProps from 'react-fake-props';

import ThemedToggle from '../ThemedToggle';

describe('app/shared-components/ThemedToggle/', () => {
  describe('Snapshot test with shallow', () => {
    test('ThemedToggle should render', () => {
      const componentPath = path.join(__dirname, '../ThemedToggle.jsx');
      const props = fakeProps(componentPath);

      const component = shallow(<ThemedToggle {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
