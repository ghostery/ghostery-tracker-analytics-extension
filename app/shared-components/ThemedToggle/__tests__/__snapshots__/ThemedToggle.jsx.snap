// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/shared-components/ThemedToggle/ Snapshot test with shallow ThemedToggle should render 1`] = `
<div
  className="ThemedToggle d-flex"
>
  <div
    className="ThemedToggle__button"
    key="tooltip-sans-fake children element"
    onClick={[Function]}
  >
    <div>
      fake children element
    </div>
  </div>
</div>
`;
